DNN Platform/Library/Services/Search/Internals/LuceneControllerImpl.cs

```diff
@@ -274,41 +282,59 @@ public LuceneResults Search(LuceneSearchContext searchContext)
-            luceneResults.TotalHits = searchSecurityTrimmer.TotalHits;
-
-            if (Logger.IsDebugEnabled)
+            try
             {
-                var sb = GetSearcResultExplanation(searchContext.LuceneQuery, searchSecurityTrimmer.ScoreDocs, searcher);
-		        Logger.Trace(sb);
-            }
+                var searcher = GetSearcher();
+                var searchSecurityTrimmer = new SearchSecurityTrimmer(new SearchSecurityTrimmerContext
+                    {
+                        Searcher = searcher,
+                        SecurityChecker = searchContext.SecurityCheckerDelegate,
+                        LuceneQuery = searchContext.LuceneQuery,
+                        SearchQuery = searchContext.SearchQuery
+                    });
+                searcher.Search(searchContext.LuceneQuery.Query, null, searchSecurityTrimmer);
+                luceneResults.TotalHits = searchSecurityTrimmer.TotalHits;
+
+                if (Logger.IsDebugEnabled)
+                {
+                    var sb = GetSearcResultExplanation(searchContext.LuceneQuery, searchSecurityTrimmer.ScoreDocs, searcher);
+		            Logger.Trace(sb);
+                }
+
+                //Page doesn't exist
+                if (luceneResults.TotalHits < minResults)
+                    return luceneResults;
 
-            //Page doesn't exist
-            if (luceneResults.TotalHits < minResults)
+                luceneResults.Results  = searchSecurityTrimmer.ScoreDocs.Select(match =>
+                    new LuceneResult
+                        {
+                            Document = searcher.Doc(match.Doc),
+                            Score = match.Score,
+                            DisplayScore = GetDisplayScoreFromMatch(match.ToString()),
+                            TitleSnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.TitleTag, searchContext.LuceneQuery.TitleSnippetLength),
+                            BodySnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.BodyTag, searchContext.LuceneQuery.BodySnippetLength),
+                            DescriptionSnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.DescriptionTag, searchContext.LuceneQuery.TitleSnippetLength),
+                            TagSnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.Tag, searchContext.LuceneQuery.TitleSnippetLength),
+                            AuthorSnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.AuthorNameTag, searchContext.LuceneQuery.TitleSnippetLength),
+                            ContentSnippet = GetHighlightedText(highlighter, fieldQuery, searcher, match, Constants.ContentTag, searchContext.LuceneQuery.TitleSnippetLength)
+                        }).ToList();
                 return luceneResults;
+            }
+            catch (Exception ex)
```

Comment:
If possible, please catch a more specific exception. You mention `IOException` in the ticket. Any other exception probably should not cause a retry, right?

Reply:
yep, there have more exceptions but not only this one after deploy the patch and get the error log, have create several sub tasks to handle these errors.

Reply:
Then please add a specific `catch` clause for each "expected exception". Let's start by catching the IOExcpetion in this PR and subsequent submissions will add more handling